tools/run_block.py

```diff
@@ -118,11 +110,11 @@ def run_generator(
         _, asset_id, _ = curried_args
         memo = ""
 
-        result = puzzle.run(solution)
+        puzzle_result = puzzle.to_program().run(solution)
```

Comment:
as far as I can tell, this will de-serialize the program (`puzzle` is a `SerizliedProgram`, right?) and then immediately serialize it again, to run it